Dockerfile

```diff
@@ -31,20 +31,18 @@ RUN set -x; yum -y update && \
 		gcc gcc-c++ make bison flex file systemd-devel zlib-devel gtest-devel rpm-build java-11-openjdk-devel \
 		expect rpm-sign zip wget && \
 		yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm && \
-		yum install -y cmake3 && \
-		ln -fs cmake3 /usr/bin/cmake && \
-		# Download and install CMake manually (zypper only gives us CMake 3.5, but we need >= 3.12)
-		wget https://github.com/Kitware/CMake/releases/download/v3.25.2/cmake-3.25.2-linux-x86_64.sh -O cmake.sh && \
-		(echo '4d98de8d605da676e71a889dd94f80c76abb377fade2f21e3510e62ece1e1ada  cmake.sh' | sha256sum --check) && \
-		bash cmake.sh --skip-license --prefix=/usr/local
+		wget https://github.com/Kitware/CMake/releases/download/v3.25.2/cmake-3.25.2-linux-$(uname -m).sh -O cmake.sh && \
```

Comment:
This is `$basearch`, which, as I recall, is `$(uname -i)` (see, e.g., https://www.linuxquestions.org/questions/linux-software-2/yum-variables-%24distribution-%24releasever-%24basearch-how-to-check-648050/). Also, you need to quote the command substitution.

But this would also prevent any sort of cross-build in the future, so maybe parameterize the architecture as a build argument instead?

Reply:
`uname -i` returns `unknown` on my x86_64 instances :(

Parameterizing with a build argument would also resolve the `ADD` -> `RUN` issue; I wanted to avoid that complexity but if it helps here too then maybe there's enough upsides to go through with it. I'll give it a try.

Reply:
I think it's worth trying to parametrize this as well, as the ADD -> RUN is a bit painful. If so, it might be worth it to add this to the template itself and make it so every build will build cmake in this way to make things consistent

Reply:
Changed to `BUILDARCH` which automatically populates with `amd64`/`arm64` when using `DOCKER_BUILDKIT=1`.